### ISSUE TITLE
Training file

### ISSUE THREAD

**Issue by vitallalaji** [google-research/tapas#19]
Hi, I want to train sqa model with maximum position embeddings 2048..so to train the model which file or command needs to be run ? Where can I change position embeddings values ?

**Comment by muelletm** [comment]
- An example command for training on SQA is given in the [readme](https://github.com/google-research/tapas).
- The data is also linked in the data section.
- Changing the embedding dimension is discussed in #14.
- You also want to change the sequence length for data creation and training (`--max_seq_length=2048`).

Does that help?

**Comment by vitallalaji** [comment]
Tq

**Comment by vitallalaji** [comment]
What is the system configuration required for training ?

**Comment by muelletm** [comment]
What do mean by system configuration?

We have never tried this the maximal length we tried so far was 1024. For 1024 we were able to train with a batch size of 128 (using 32 Cloud TPU v3).

If you want to train this on GPU, I would recommend to start with a BERT base model instead of BERT large and see if there is any batch size that works but it would probably be very small you should also look into gradient accumulation.

But in general it will be hard to get this to work. Maybe there is an easy way to discard some irrelevant rows / columns?

**Comment by vitallalaji** [comment]
What is the equivalent configuration for AWS for cloud 32 tpu v3 ??

**Comment by muelletm** [comment]
Only Google Cloud offers TPUs. For AWS you would have to go for a GPU setup.

You will have to reduce the batch size, though. [BERT](https://github.com/google-research/bert#out-of-memory-issues) lists the batch sizes that are possible for certain sequence lengths on a Titan X. TAPAS would have the same requirements. As you can see they don't specify what is needed for 1024 and 2048 and it could be impossible to train such big models on GPUs.

Therefore you should first consider if you can discard some irrelevant cells or possible split your table into multiple smaller tables.

**Comment by vitallalaji** [comment]
Can you make a vedio on explaining tapas_classifier _model code ?? The coding part is hard to understand...

**Comment by muelletm** [comment]
Sorry, I don't think we can release a video. Do you have specific questions? Maybe we can expand the comments or provide some explanation.
